src/main/java/org/junit/runners/Parameterized.java

```diff
@@ -276,10 +330,15 @@ private ParametersRunnerFactory getParametersRunnerFactory()
             }
         }
 
+        private Integer parameterCount;
+
         private TestWithParameters createTestWithNotNormalizedParameters(
                 String pattern, int index, Object parametersOrSingleParameter) {
             Object[] parameters = (parametersOrSingleParameter instanceof Object[]) ? (Object[]) parametersOrSingleParameter
                     : new Object[] { parametersOrSingleParameter };
+            if (parameterCount == null) {
+                parameterCount = parameters.length;
+            }
```

Comment:
As a side effect we could have a check that number of parameters is always the same.